pkg/planner/core/point_get_plan.go

```diff
@@ -1527,8 +1566,56 @@ func checkTblIndexForPointPlan(ctx base.PlanContext, tblName *resolve.TableNameW
 
 // indexIsAvailableByHints checks whether this index is filtered by these specified index hints.
 // idxInfo is PK if it's nil
-func indexIsAvailableByHints(idxInfo *model.IndexInfo, idxHints []*ast.IndexHint) bool {
-	if len(idxHints) == 0 {
+func indexIsAvailableByHints(
+	currentDB string,
+	dbName string,
+	tblAlias string,
+	idxInfo *model.IndexInfo,
+	tblHints []*ast.TableOptimizerHint,
+	idxHints []*ast.IndexHint,
+) bool {
+	combinedHints := idxHints
+
+	// Handle *ast.TableOptimizerHint, which is from the comment style hint in the SQL.
+	// In normal planner code path, it's processed by ParsePlanHints() + getPossibleAccessPaths().
+	// ParsePlanHints() converts the use/force/ignore index hints in []*ast.TableOptimizerHint into *ast.IndexHint, and
+	// put it into *hint.PlanHints.
+	// getPossibleAccessPaths() receives the *hint.PlanHints, matches the table names, and combines the hints with the
+	// USE/IGNORE/FORCE INDEX () syntax hint.
+	// Here we try to remove the logic that is not needed in the fast path, and implement a faster and simpler version
+	// to fit the fast path.
+	for _, h := range tblHints {
+		var hintType ast.IndexHintType
+		// The parsing logic from ParsePlanHints()
+		switch h.HintName.L {
+		case hint.HintUseIndex:
+			hintType = ast.HintUse
+		case hint.HintIgnoreIndex:
+			hintType = ast.HintIgnore
+		case hint.HintForceIndex:
+			hintType = ast.HintForce
+		default:
+			continue
+		}
+		if len(h.Tables) != 1 {
```

Comment:
May I ask when this is expected to happen?

Reply:
I just checked the .y definition. It should not happen. I also added an intest assert here.

https://github.com/pingcap/tidb/blob/b2f2faaa95bff642920560c24e3714957bcb4c74/pkg/parser/hintparser.y#L521-L555